### PR TITLE
forcing local repo checks for 'file://' prefixed repositories - issues/715

### PR DESCRIPTION
**Pull request by jbkc85**

### CODE REVIEW

**Comment by fiftin** [comment]
Thank you @jbkc85 !

**Comment by jbkc85** [comment]
> Thank you @jbkc85 !

not a problem @fiftin - I hope there was a way to test this out...I still have yet to be able to get all the javascript stuff up and running on my local machine :-(

File: api/tasks/runner.go [hunk 1]
```diff
@@ -173,10 +175,17 @@ func (t *task) prepareRun() {
 		return
 	}
 
-	if err := t.updateRepository(); err != nil {
-		t.log("Failed updating repository: " + err.Error())
-		t.fail()
-		return
+	if strings.HasPrefix(t.repository.GitURL, gitURLFilePrefix) {
+		repositoryPath := strings.TrimPrefix(gitURLFilePrefix, t.repository.GitURL)
```
Comment (cameronpm) [inline api/tasks/runner.go:203]: Wrong way round, the prefix is the 2nd argument

https://pkg.go.dev/strings#TrimPrefix

File: api/tasks/runner.go [hunk 2]
```diff
@@ -173,10 +175,17 @@ func (t *task) prepareRun() {
...
+		if _, err := os.Stat(repositoryPath); err != nil {
+			t.log("Failed in finding static repository at " + repositoryPath + ": " + err.Error())
```
Comment (cameronpm) [inline api/tasks/runner.go:205]: Should fail() and return here ?

File: api/tasks/runner.go (hunk 1 above)
Comment (jbkc85) [reply api/tasks/runner.go:203]: ugh, seriously?  I thought I had that appropriately set.

File: api/tasks/runner.go (hunk 2 above)
Comment (jbkc85) [reply api/tasks/runner.go:205]: Yes.  It should.

**Pull request merged**